### PR TITLE
Support for multi select and adding new values to select.

### PR DESCRIPTION
**Pull request by tittarii**
Issue #98

### CODE REVIEW

File: src/components/Tos/DropdownInput/DropdownInput.js [hunk 1]
```diff
@@ -81,28 +96,39 @@ export const DropdownInput = ({
         });
       }
     }
+    const valueArray = valueState instanceof Array ? valueState : [valueState];
+    forEach(valueArray, function(value) {
+      if (!find(optionsArray, function(option) { return option.value === value; })) {
```
Comment (jussiarpalahti) [inline src/components/Tos/DropdownInput/DropdownInput.js]: These should be abstracted away into function that's named for what the check is about.

I don't quite get what is happening here, but having names for checks would help.

File: src/utils/validators.js [hunk 2]
```diff
@@ -54,7 +63,17 @@ export const validateTOS = (tos, rules) => {
  */
 export const validatePhase = (phase, rules) => {
   const errors = [];
-  // TODO: implementation
+  for (const key in rules) {
+    if (phase.attributes[key] && rules[key].values.length) {
```
Comment (jussiarpalahti) [inline src/utils/validators.js]: Same goes for these constructs. Having functions do the checking does add layer of indirection, but helps in reading.

**Comment by jussiarpalahti** [comment]
@tittarii, I added some comments. It seems like same data here can be either one item or many. Perhaps it's necessary for the way app works.

**Comment by tittarii** [comment]
This issue was to enable user to add his own values even there are predefined values for select elements. If value is not found in predefined options, the value is added to options list for the select ui component to show (DropdownInput).

Multiselection support was added, so value can be String or Array. In validation value(s) is checked that it is a valid option (not user written value). 

I did some refactoring to make code more readable. Thanks for the feedback!

**Comment by jussiarpalahti** [comment]
Ok, thanks. Merging.

**Pull request merged**